build.cmd

```diff
@@ -6,10 +6,20 @@ setlocal
 ::       means that that rebuilding cannot successfully delete the task
 ::       assembly.
 
+:ReadArguments
+:: Read in the args to determine whether to run the native build, managed build, or both (default)
+set "__args=%*"
+set "__buildSpec=both"
+if /i [%1] == [both]    	(set __buildSpec=both&&set "__args=%__args:~4%"&&shift&&goto Tools)
+if /i [%1] == [native]    	(set __buildSpec=native&&set "__args=%__args:~6%"&&shift&&goto Tools)
+if /i [%1] == [managed]    	(set __buildSpec=managed&&set "__args=%__args:~7%"&&shift&&goto Tools)
```

Comment:
Perhaps I'm missing something, but shouldn't all we need is the shift to remove the argument and not have to use the __args env var?